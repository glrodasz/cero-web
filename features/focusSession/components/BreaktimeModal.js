import PropTypes from 'prop-types'

import {
  Modal,
  CenteredContent,
  Picture,
  Heading,
  Spacer,
  Paragraph,
  Button,
} from '@glrodasz/components'

const BreaktimeModal = ({ onClickClose }) => {
  return (
    <Modal isCentered onClose={onClickClose}>
      <CenteredContent>
        <Picture src="/images/couch-pause.svg" width={200}></Picture>
        <Heading size="xl" color="tertiary" isInline>
          Tomate un tiempo para refrescarte
        </Heading>
        <Spacer.Horizontal size=" sm" />
        <Paragraph>
          Siempre hay que celebrar los peque√±os triunfos, por eso te invitamos a
          tomar un descanso para despejar tu mente.
        </Paragraph>
        <Spacer.Horizontal size="md" />
        <div style={{ display: 'flex', gap: '0 20px' }}>
          <Button isMuted>5 min</Button>
          <Button isMuted>10 min</Button>
          <Button isMuted>15 min</Button>
        </div>
      </CenteredContent>
    </Modal>
  )
}

BreaktimeModal.propTypes = {
  onClickClose: PropTypes.func.isRequired,
}

export default BreaktimeModal
