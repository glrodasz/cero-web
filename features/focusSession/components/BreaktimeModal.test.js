import BreaktimeModal from './BreaktimeModal'
import { render } from '@testing-library/react'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../utils/testing')
  return {
    Modal: dummyRender('Modal'),
    CenteredContent: dummyRender('CenteredContent'),
    Picture: dummyRender('Picture'),
    Heading: dummyRender('Heading'),
    Spacer: { Horizontal: dummyRender('Spacer') },
    Paragraph: dummyRender('Paragraph'),
    Button: dummyRender('Button'),
  }
})

describe('[ features / focusSession / components / BreaktimeModal ]', () => {
  describe('when `BreaktimeModal` is mounted', () => {
    it('should render', () => {
      const props = {
        onClickClose: () => {},
      }
      const { asFragment } = render(<BreaktimeModal {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
