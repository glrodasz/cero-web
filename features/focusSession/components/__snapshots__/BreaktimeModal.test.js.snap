// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / focusSession / components / BreaktimeModal ] when \`BreaktimeModal\` is mounted should render 1`] = `
<DocumentFragment>
  [Modal {"isCentered":true}][CenteredContent][Picture {"src":"/images/couch-pause.svg","width":200} /] [Heading {"size":"xl","color":"tertiary","isInline":true}]Tomate un tiempo para refrescarte[/Heading] [Spacer {"size":" sm"} /] [Paragraph]Siempre hay que celebrar los peque√±os triunfos, por eso te invitamos a tomar un descanso para despejar tu mente.[/Paragraph] [Spacer {"size":"md"} /] [div style="display:flex;gap:0 20px"][Button {"isMuted":true}]5 min[/Button] [Button {"isMuted":true}]10 min[/Button] [Button {"isMuted":true}]15 min[/Button] [/div][/CenteredContent] [/Modal] 
</DocumentFragment>
`;
